src/hotrod/impl/TimeUnitParam.h

```diff
@@ -0,0 +1,24 @@
+#ifndef INFINISPAN_HOTROD_CPP_TIMEUNITPARAM_H
+#define INFINISPAN_HOTROD_CPP_TIMEUNITPARAM_H
+
+#include "infinispan/hotrod/TimeUnit.h"
+
+namespace infinispan {
+namespace hotrod {
+    class TimeUnitParam {
+        static TimeUnit encodeDuration(long duration, TimeUnit timeUnit) {
+            if (duration == 0) return DEFAULT;
+            if (duration < 0) return INFINITE;
+            return timeUnit;
```

Comment:
this function evaluates only the measurement unit (4 bits). The result should be the same of the java version TimeUnitParam.encodeDuration().